cng/cng.go

```diff
@@ -97,3 +98,26 @@ func getUint32(h bcrypt.HANDLE, name string) (uint32, error) {
 	err := bcrypt.GetProperty(h, utf16PtrFromString(name), (*[4]byte)(unsafe.Pointer(&prop))[:], &discard, 0)
 	return prop, err
 }
+
+func getKeyLengths(h bcrypt.HANDLE) (lengths bcrypt.KEY_LENGTHS_STRUCT, err error) {
+	var discard uint32
+	err = bcrypt.GetProperty(bcrypt.HANDLE(h), utf16PtrFromString(bcrypt.KEY_LENGTHS), (*[unsafe.Sizeof(lengths)]byte)(unsafe.Pointer(&lengths))[:], &discard, 0)
+	if err != nil {
+		return
+	}
+	if lengths.MinLength > lengths.MaxLength || (lengths.Increment == 0 && lengths.MinLength != lengths.MaxLength) {
+		err = errors.New("invalid BCRYPT_KEY_LENGTHS_STRUCT")
+		return
+	}
+	return lengths, nil
+}
+
+func keyIsAllowed(lengths bcrypt.KEY_LENGTHS_STRUCT, bits uint32) bool {
+	if bits < lengths.MinLength || bits > lengths.MaxLength {
+		return false
+	}
+	if lengths.Increment == 0 {
+		return bits == lengths.MinLength
```

Comment:
Interesting. Poking around, I see a few places that have a 0-increment test case (where min == max), so if that actually does happen in the wild, this PR is also fixing support for this value, where previously we would error out. 🎉

https://github.com/dotnet/pinvoke/blob/cc452f8f1d9437827b8181b06a153c6fd5997d08/src/BCrypt.Tests/BCryptFacts.cs#L65-L70
https://github.com/dotnet/aspnetcore/blob/5834214ddeab320106c8e728b92b611a8c6add63/src/DataProtection/Cryptography.Internal/test/Cng/BCRYPT_KEY_LENGTHS_STRUCT_Tests.cs#L13-L17